include/bluetooth/bluetooth.h

```diff
@@ -681,6 +724,9 @@ struct bt_le_scan_cb {
 	void (*recv)(const struct bt_le_adv_info *info,
 		     struct net_buf_simple *buf);
 
+	/** @brief The scanner has stopped scanning after scan timeout. */
+	void (*timeout)(void);
```

Comment:
I wonder if we could just use `recv()` with `info == NULL`. It's just more and more code piling up otherwise.

Reply:
Not a big fan of that idea. I find timeout to be much cleaner. It's 2 different HCI events as well. So I don't think there is much code to be saved here either.